src/index.ts

```diff
@@ -13,49 +13,80 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
 import * as extend from 'extend';
 import * as path from 'path';
-import {AuthenticationConfig, Common, ServiceConfig} from '../third_party/types/common-types';
-import {Config, defaultConfig} from './config';
+const gcpMetadata = require('gcp-metadata');
+
+import {AuthenticationConfig, Common, ServiceConfig, ServiceObject} from '../third_party/types/common-types';
+
+import {Config, defaultConfig, internalConfig} from './config';
 import {Profiler, ProfilerConfig} from './profiler';
 
 const common: Common = require('@google-cloud/common');
+const metadataAPI = 'http://metadata.google.internal/computeMetadata/v1/';
+
+// Returns value of metadata field.
+// Throws error if there is a problem accessing metadata API.
+function getField(field: string): Promise<string> {
```

Comment:
You can replace this function by using`pify` and its `multiArgs` option: https://www.npmjs.com/package/pify#multiargs